prime-router/src/main/kotlin/Element.kt

```diff
@@ -527,21 +527,15 @@ data class Element(
                     // continue to the next try
                 }
                 try {
-                    // this is a saving throw
-                    val optionalDateTime = variableDateTimePattern
-                    val df = DateTimeFormatter.ofPattern(optionalDateTime)
-                    val ta = df.parseBest(
-                        cleanedValue,
-                        OffsetDateTime::from,
-                        LocalDateTime::from,
-                        Instant::from,
-                        LocalDate::from
-                    )
-                    if (ta is LocalDateTime) {
-                        LocalDateTime.from(ta).atZone(ZoneId.of(USTimeZone.CENTRAL.zoneId)).toOffsetDateTime()
-                    } else {
-                        LocalDate.from(ta).atStartOfDay(ZoneId.of(USTimeZone.CENTRAL.zoneId)).toOffsetDateTime()
-                    }
+                    getBestDateTime(cleanedValue, variableDateTimePattern)
```

Comment:
This looks like duplicate code line in line 714.  Is there a possibility of refactoring this code into a function then reusing?

Reply:
Done.